### DISCUSSION TITLE
What do you think of using Tailwind UI components?

### DISCUSSION THREAD

**Discussion by ykdojo** [ykdojo/defaang#98] [General]
I was thinking about how we can make this project look professional, clean, and consistent across the board especially as we have more community contributions.

Then, I came across [Tailwind UI](https://tailwindui.com/).

What do you think about using it?

I looked for alternatives, but none of the other ones I found looked as good as this one.

I can pay for it if necessary.

**Comment by rohitdasu** [comment]
Hi @ykdojo i've previously used Tailwind UI. It's nice and flexible. We can modify as we want.

**Comment by yaa-chi** [comment]
> I was thinking about how we can make this project look professional, clean, and consistent across the board especially as we have more community contributions.
> 
> Then, I came across [Tailwind UI](https://tailwindui.com/).
> 
> What do you think about using it?
> 
> I looked for alternatives, but none of the other ones I found looked as good as this one.
> 
> I can pay for it if necessary.

Actually, I think tailwind css would be great when it comes to making the defaang website look more professional and cleen

**Reply by ykdojo** [reply]
We're already using Tailwind CSS! Tailwind UI is a component library on top of it.

**Reply by yaa-chi** [reply]
Really? 😂🔥 I had not idea!

**Comment by subhoghoshX** [comment]
Definitely, we should go for it. I was going to suggest it, but then I saw people making their own designs, saw I hold myself back.

I think the free components are already good. But premium ones would be awesome if you wanna pay for them.

**Reply by ykdojo** [reply]
Right. I think we can replace the existing things with it over time, too.

**Reply by yaa-chi** [reply]
Does this ui component give you the ability to design custom designs? Like Bootstrap?

**Reply by rohitdasu** [reply]
Yes it provides components like bootstrap but these are flexible, means they can be designed as per our requirement. Sky is the limit 😆

**Reply by rohitdasu** [reply]
Bootstrap components can be hardly modifiable but TailwindUI components can be modified easily

**Reply by subhoghoshX** [reply]
Yes. Tailwind CSS components are built by applying small utility classes to HTML. You can just change those classes (margin, padding, size, color, borders) to customize it as your will.

**Reply by yaa-chi** [reply]
Woooow, then thats greattt! So we can actually have more freedom with our designs.

@ykdojo please answer me quickly concerning the color of the texts so that we can start designing asap😂🔥

**Comment by veganaiZe** [comment]
In all honesty, I think it looks cheap & generic.  I feel like I could make something equal or better :icecream: _--but I wouldn't blame you if you're getting sick of hearing me say that!_

I'll support whatever decision is made.

**Reply by ykdojo** [reply]
To me, one of the main challenges here is consistency across the board.

When someone tries to contribute their own UI design, it's hard to keep it consistent with everything else.

I guess we could have someone check everything and fix every PR, but that seems pretty time-consuming. Starting with Tailwind UI will give us a good floor, at least.

**Comment by kaleem68** [comment]
I have an alternative suggestion https://chakra-ui.com/
I pretty much use Chakra UI all the time. It's a great alternative. What do you all think?

**Reply by ykdojo** [reply]
Does it work with Tailwind?

**Reply by kaleem68** [reply]
You can think Chakra UI is an alternative to Tailwind CSS. Here is a good comparison

How is Chakra different from Tailwind CSS?

https://chakra-ui.com/getting-started/comparison


If you read it, I think the key benefit of using Chakra are:
  - Accessibility
  - Responsiveness

**Reply by Dhruv-Gajjar** [reply]
I have used chakra Ui as well as tailwindcss i think chakraUi is more like react based approach like props, but i think both are a bit different though.

**Reply by kaleem68** [reply]
Chakra UI also has support for nextjs https://chakra-ui.com/getting-started/nextjs-guide if that is what we need.

**Comment by mathiasayivor** [comment]
Using Tailwind UI imo, provides util components like dialogs, alerts, cards, form controls, etc out of the box.
But sometimes, using a component library may limit your ability to create custom UI/styles...

**Reply by yaa-chi** [reply]
not really, you can still tweak them. I do have some experience in that for front end so I'll check out tailwind CSS library and see👌🏾✨

**Reply by subhoghoshX** [reply]
Yes, you can tweak them as much as you want. It's more customizable than say bootstrap.

Unlike traditional components, in Tailwind UI the markup and the style information (i.e. utility classes) are all in the same file, so it's much easier and safer to change.

If you guys need any help just let me know.
